frontend/app/assets/javascripts/spree/frontend/views/spree/home/product_carousels.js

```diff
@@ -9,23 +9,30 @@ Spree.fetchProductCarousel = function (taxonId, htmlContainer) {
   })
 }
 
+Spree.loadCarousel = function (element, div) {
+  var container = $(element)
+  var productCarousel = $(div)
+  var carouselLoaded = productCarousel.attr('data-product-carousel-loaded')
+
+  if (container.length && !carouselLoaded && container.isInViewport()) {
+    var taxonId = productCarousel.attr('data-product-carousel-taxon-id')
+    productCarousel.attr('data-product-carousel-loaded', 'true')
+
+    Spree.fetchProductCarousel(taxonId, container)
+  }
+}
+
 document.addEventListener('turbolinks:load', function () {
   var homePage = $('body#home')
 
   if (homePage.length) {
-    $(window).on('resize scroll', function () {
-      $('div[data-product-carousel').each(function (_index, element) {
-        var container = $(element)
-        var productCarousel = $(this)
-        var carouselLoaded = productCarousel.attr('data-product-carousel-loaded')
+    var carousels = $('div[data-product-carousel')
+    // load Carousels straight away if they are in the viewport
+    carousels.each(function (_index, element) { Spree.loadCarousel(element, this) })
```

Comment:
lines 31 and 35 are exactly the same `carousels.each(function (_index, element) { Spree.loadCarousel(element, this) })`. Could you refactor this?

Reply:
Sure thanks for the comment.